### ISSUE TITLE
Convert experimental env into globalOnly `experimentalFlags`

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#27879]
### Describe the proposed change(s).

Remove all `RENOVATE_X_` logic in code, convert instead to a new admin option `experimentalFlags`. Convert docs accordingly (all should be documented and it enforced with type checking).

**Comment by RahulGautamSingh** [comment]
There are some variables in this list which don't have the prefix `RENOVATE_X`, should they be included in this refactor too?

**Comment by RahulGautamSingh** [comment]
Do you mean we store all these variables and values in `experimentalFlags` object? 
ie.,
_old_
```env
# .env
RENOVATE_X_HARD_EXIT=false
```

_new_
```js
// config.js
"exprimentalFlags": {
  "HARD_EXIT": false
}
```

**Comment by rarkins** [comment]
I was going to keep them as an array. If they need a value then it could be like abc=10. One like this could be turned around such as "softExit" or "exitCodeZero"

**Comment by RahulGautamSingh** [comment]
There are some options such as `RENOVATE_X_IGNORE_RE2` which are used before `GlobalConfig` is initialized, what should be do about them?

**Comment by rarkins** [comment]
How many of those are there? I guess we need to keep them

**Comment by RahulGautamSingh** [comment]
Two with the prefix `RENOVATE_X`:

 `RENOVATE_X_HARD_EXIT`          https://github.com/renovatebot/renovate/blob/bf91e94b4b95a2a9274319aeaae8e03f05605258/lib/renovate.ts#L22        
 
 `RENOVATE_X_IGNORE_RE2`         https://github.com/renovatebot/renovate/blob/bf91e94b4b95a2a9274319aeaae8e03f05605258/lib/util/regex.ts#L18

**Comment by rarkins** [comment]
Ok keep them as is for now

**Comment by RahulGautamSingh** [comment]
I've put together [this ](https://github.com/renovatebot/renovate/pull/28187/)PR as a starting point. I've tried to cover some of the major aspects:

1. Sorting out the validation logic
2. Adding documentation for things like experimentalFlags and a few other flags. The idea is to set up a template for handling similar cases down the line.
3. I've also included a utility function. It helps with pulling values out of the experimentalFlags array. 
For example, if we have the following config  this function makes sure we can easily get the value when needed.
``` experimentalFlags = ['dockerHubTags', 'dockerMaxPages=10'] ```

I have also commented on parts that I am not very sure of. We can delve into the details there, and once sorted, we'll only have the conversion part to tackle.

**Comment by secustor** [comment]
IMO it make sense to have `experimental` in the naming. 
We communicate that these features are unstable and subject to change without looking at the docs. 
Further that extra scrutiny is expected as they are not part of the normal configuration. 

If that is not the intend we can migrate all flags to config options with the experimental setting. 🤷

**Comment by rarkins** [comment]
ok

**Comment by RahulGautamSingh** [comment]
Fresh list of the the experimental vars which remain to be converted, with some related infromation.

| Env Var                                          | Conversion Possible | Remarks                                                                                                   | Possible Config Option Name        |
| ------------------------------------------------ | ------------------- | --------------------------------------------------------------------------------------------------------- | ---------------------------------- |
| `OTEL_EXPORTER_OTLP_ENDPOINT`                    | ❌                   | Used before global config class is initialized                                                            |                                    |
| `RENOVATE_CACHE_NPM_MINUTES`                     | ✅                   |                                                                                                           | `cacheNpmMinutes`                  |
| `RENOVATE_X_DELETE_CONFIG_FILE`                  | ✅                   |                                                                                                           | `deleteConfigFile`                 |
| `RENOVATE_X_DOCKER_MAX_PAGES`                    | ✅                   |                                                                                                           | `dockerMaxPages`                   |
| `RENOVATE_X_EAGER_GLOBAL_EXTENDS`                | ✅                   |                                                                                                           | `eagerGlobalExtends`               |
| `RENOVATE_X_GITLAB_AUTO_MERGEABLE_CHECK_ATTEMPS` | ✅                   |                                                                                                           | `gitlabAutoMergeableCheckAttempts` |
| `RENOVATE_X_GITLAB_BRANCH_STATUS_DELAY`          | ✅                   |                                                                                                           | `gitlabBranchStatusDelay`          |
| `RENOVATE_X_GITLAB_MERGE_REQUEST_DELAY`          | ✅                   |                                                                                                           | `gitlabMergeRequestDelay`          |
| `RENOVATE_X_HARD_EXIT`                           | ✅                   |                                                                                                           | `hardExit`                         |
| `RENOVATE_X_IGNORE_NODE_WARN`                    | ✅                   |                                                                                                           | `ignoreNodeWarn`                   |
| `RENOVATE_X_IGNORE_RE2`                          | ❌                   | Regex is used at one too many places to be sure, if it will be called before global config is initialized |                                    |
| `RENOVATE_X_PLATFORM_VERSION`                    | ✅                   |                                                                                                           | `platformVersion`                  |
| `RENOVATE_X_S3_ENDPOINT`                         | ✅                   |                                                                                                           | `s3Endpoint`                       |
| `RENOVATE_X_S3_PATH_STYLE`                       | ✅                   |                                                                                                           | `s3PathStyle`                      |

**Comment by rarkins** [comment]
No. The main reason this feature was made opt-in was because it doesn't work with app tokens
